src/protocols/bdx/BDXMessageUtils.h

```diff
@@ -117,18 +125,27 @@ struct SendAccept
      *
      * @return CHIP_ERROR Any error that occurs when trying to write to the PacketBuffer
      */
-    CHIP_ERROR Pack(System::PacketBuffer * aBuffer);
+    CHIP_ERROR Pack(System::PacketBuffer & aBuffer);
 
     /**
      * @brief
      *  Parse data from an PacketBuffer into a struct instance
      *
      * @param[in] aBuffer Pointer to a PacketBuffer containing the data
-     * @param[out] aParsedMessage Reference to a struct instance where results will be stored
+     * @param[out] aParsedMessage Reference to a struct instance where results will be stored.
+     *             Note that this struct will store pointers into the passed PacketBuffer, so it
+     *             is essential that the PacketBuffer is not modified or freed until after the
+     *             SendAccept struct is no longer needed.
      *
      * @return CHIP_ERROR Any error that occurs when trying to read the message
      */
-    static CHIP_ERROR Parse(System::PacketBuffer * aBuffer, SendAccept & aResponse);
+    static CHIP_ERROR Parse(System::PacketBuffer & aBuffer, SendAccept & aResponse);
```

Comment:
This should be `const System::PacketBuffer & aBuffer`, because aBuffer is read in place and not mutated. If you can't make it a const reference, then it means the PacketBuffer API needs to be tightened for correct constness.

Reply:
I'll review the methods again and mark anything const that should be